assets/javascripts/kitten/components/uploaders/document-manager/index.js

```diff
@@ -295,12 +295,11 @@ export const DocumentManager = ({
           )}
         >
           <div className="k-DocumentManager__uploader__container">
-            <div className="k-DocumentManager__uploader__documentIcon">
+            <div className="k-DocumentManager__uploader__documentIcon k-u-hidden@xs-down">
```

Comment:
Que penses-tu d'ajouter la gestion du responsive via `k-DocumentManager__uploader__documentIcon` ou une autre classe propre au composant ? L'utilisation des helpers dans un composant React impose à l'application de charger les utilitaires. C'est pas forcement le cas.

Reply:
> C'est pas forcement le cas.

En l'état, c'est avec les helpers que le composant `Text` fonctionne depuis le début. Le chargement des utilitaires est requis pour que le composant le plus basique de Kitten fonctionne.

Reply:
C'est pas faux. J'ai l'impression que `Text` reste quand même un composant à part. Il se base à 100% sur les utilitaires. Dans le cas ici, j'ai quand même l'impression que moins on ajoute de dépendances externes mieux c'est.

La remarque reste non-bloquante si on part du principe que toutes les apps qui utilisent Kitten auront forcement les utilitaires.

Reply:
@LanF3usT du coup j'ai appliqué ta proposition

Reply:
C'est vraiment dommage de ne pas utiliser les utilitaires dans Kitten. Ça retire des composants utilisables par les gens qui n'incluent pas les utilitaires (on ne sait pas s'ils existent), ça apporte des problèmes pour des utilisateurs qui existent (nous) :

- Personne (dont on est au courant) n'utilise Kitten sans les utilitaires. Ces personnes là (si elles existent) ne peuvent pas utiliser :
  - `Text`
  - Tous les composants qui utilisent `Text` (soit une 30aine)
  - Tous les composants qui utilisent des utilitaires (pas encore compté)
- Les personnes qui incluent les utilitaires (KissKiss et Lendo) ont une baisse de performance :
  - On déclare plusieurs fois des définitions quand on pourrait les avoir à un seul endroit, ce qui alourdit notre bundle JS
  - Beaucoup des utilitaires SCSS sont remplacés par des règles CSS dans les composants, qui utilisent des calculs JS pour être rendus (quand tu insères `${TYPOGRAPHY.fontStyles.light}` tu exécutes du JS, quand tu fais une `@media (min-width: ${pxToRem(ScreenConfig.M.min)})` tu exécutes du JS, etc)

Au niveau principes de développement, ça va aussi à l'encontre de ce qui me parait plus logique :

- On rend un service à des développeurs qui n'existent pas (ceux qui utilisent Kitten sans vouloir charger les utilitaires SCSS)
- On ne met pas en commun le code qui peut l'être
- On se prive d'outils qui ont été développés dans cette optique spécifique

Reply:
Je n'ai pas d'opinion forte sur la question, @LanF3usT, qu'en penses-tu ?

Reply:
Je reste fidèle à mon opinion vis à vis des composants Kitten. Ma vision `idéale` d'un composant Kitten est qu'il doit s'auto-suffire en tout point. Il n'est pas censé dépendre d'éléments externes (classes CSS, reset CSS, etc.). Le simple import d'un composant Kitten dans une app, on est censé avoir le même rendu partout. Comme tu le soulignes à juste titre, on est très loin d'avoir cela actuellement pour nos composants. 

Le côté open-source de la librairie nous impose des choix qui nous dépassent parfois. La frontière d'un breaking change est mince. Mon avis par rapport à ca, est qu'il faut tout le temps se dire que Kitten est utilisé par d'autres personnes/apps (même si dans les faits je ne pense pas). On a jamais remis en question le côté open-source de la lib mais si on sent que c'est un frein pour nos devs, on peut envisager d'avoir un Kitten privé. Je pense que ca peut avoir du sens surtout pour pouvoir breaking change sans contrainte. Et pourquoi pas ressortir une version de Kitten en open-source plus tard quand on sent que tout est clean.

Tout cela n'est que mon avis et finalement ce n'est pas tranché. Il y a même des fois où je me dis qu'on devrait retourner dans un fonctionnement en React pour le DOM et des mixins SCSS comme avant. Le CSS-in-JS s'est fait pour faciliter l'utilisation de nos composants mais c'est pas le plus performant comme tu le dis.

Je trouve que c'est une bonne période pour tout remettre en question, de faire le bilan sur nos usages et pourquoi pas prendre une autre direction. Tout est ouvert.

Reply:
Comptez moi dans cette conversation, j'ai exploré pas mal de choses sur le sujet ces derniers temps, et ça m'intéresse beaucoup d'arriver de trouver comment garantir cette auto-suffisance d'un composant, tout en pouvant lui appliquer des thèmes, et choisir la meilleure solution technique pour ensuite appliquer son style en pratique.

Par curiosité, quels retours tu as sur les perfs du CSS-in-JS qui te font douter des performances ? Est-ce que c'est intrinsèque à ces solutions, ou est-ce que notre usage actuel n'est pas optimal ?